apps/core/serializers/comment.py

```diff
@@ -44,6 +49,9 @@ def get_content(self, obj) -> typing.Union[str, list]:
         return obj.content
 
     def get_hidden_content(self, obj) -> str:
+        if self.get_is_hidden_by_reported(obj):
+            return gettext('This comment is hidden because it received multiple reports')
+            
```

Comment:
이 함수 관련해서 질문이 있는데요, UI 상으로 원래는 게시물이 숨김처리 되어있을때 이제 내용은 안보여주고 버튼이 뜨고, 그럼에도 보겠다! 해서 버튼을 누르면 내용이 보이는 형태이잖아요,

제가 알기로 그 버튼은 why_hidden 필드가 채워져서 내려가면 프론트에서 인식하고 띄워주는 것으로 알고 있는데 < 이부분 프론트에 정확한 확인이 필요합니다.

그렇다면 신고로 의한 숨김처리는 어차피 why_hidden에는 영향이 없기 때문에 버튼이 뜨지 않을거라, get_content에서 이미 처리해주었으니 get_hidden_content 는 해당 부분이 필요가 없지 않을까요?

Reply:
:+1:
return None으로도 충분할 것 같아요.

Reply:
>제가 알기로 그 버튼은 why_hidden 필드가 채워져서 내려가면 프론트에서 인식하고 띄워주는 것으로 알고 있는데 < 이부분 프론트에 정확한 확인이 필요합니다.

아래는 new-ara-web의 src>components>PostComment.vue 의 Line 133~ 입니다. is_hidden 필드가 채워지면 다시보겠습니까 버튼이 띄워지고 why_hidden 필드의 숨김 이유가 보입니다. 그리고 버튼 클릭시 hidden_content가 보입니다.

```
    content () {
      if (this.comment.is_hidden) {
        if (this.showHidden) {
          return this.comment.hidden_content
        }

        return this.comment.why_hidden
          .map(v => v.detail.join('\n'))
          .join('\n')
      }

      return this.comment.content
    }
  },
```
제가 의도한것은 validate_hidden에 의한 숨김처리 + 신고에의한 숨김처리를 모두 당한 게시글 또는 댓글에서는 다시 보겠다는 클릭 시, hidden_content가 'This comment is hidden because it received multiple reports' 여야 한다고 생각했습니다.

Reply:
다시생각해보았는데, 주연님의 말씀대로 validate_hidden에 의한 숨김처리보다 신고에 의한 숨김처리의 비중이 크므로, serializers/comment.py의 get_is_hidden 함수를 신고에 의한 숨김처리가 된 게시글일 때는 is_hidden을 false로 전달하여 '다시보겠습니까' 버튼이 띄워지지않고 'This comment is hidden because it received multiple reports'를 보이도록 하는게 직관적인것같습니다.
그렇다면 위의 get_hidden_content는 주연님 말씀대로 필요없을 것 같습니다.

Reply:
이부분 오늘 공동코딩시간에 설명드리겠습니다.